source/common/network/utility.cc

```diff
@@ -272,12 +273,17 @@ bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
     }
 
     if (ifa->ifa_addr->sa_family == af_look_up) {
-      const auto* addr = reinterpret_cast<const struct sockaddr_storage*>(ifa->ifa_addr);
-      auto local_address = Address::addressFromSockAddr(
-          *addr, (af_look_up == AF_INET) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6));
-
-      if (remote_address == local_address) {
-        return true;
+      if (af_look_up == AF_INET) {
+        const auto* if_addr = reinterpret_cast<const struct sockaddr_in*>(ifa->ifa_addr);
+        if (remote_address->ip()->ipv4()->address() == if_addr->sin_addr.s_addr) {
+          return true;
+        }
+      } else {
+        const auto* if_addr = reinterpret_cast<const struct sockaddr_in6*>(ifa->ifa_addr);
+        const absl::uint128 remote_v6addr = remote_address->ip()->ipv6()->address();
```

Comment:
Please make sure to not overread the memory:

```
static_assert(sizeof(absl::uint128) == sizeof(if_addr->sin6_addr.s6_addr),
              "sizeof(absl::uint128) != sizeof(if_addr->sin6_addr.s6_addr)");
```

Reply:
Added.